### PR TITLE
Show Delegate Panel link to non-Delegate Board Members.

### LINKED ISSUE #3258: Non-Delegate Board Members Cannot View Delegate Panel
Board members are no longer required to be Delegates, but non-Delegate Board members do not have the link visible in the drop-down menu to view the Delegate Panel. They have access to the page by visiting the link directly, but no navigation to it.

### PR DESCRIPTION
**Pull request by AlbertoPdRF**
Fixes #3258.

### CODE REVIEW

File: WcaOnRails/app/views/layouts/_navigation.html.erb [hunk 1]
```diff
@@ -136,13 +136,13 @@
                 <li><%= link_to t('.manage_users'), users_path %></li>
               <% end %>
 
-              <% if current_user.any_kind_of_delegate? %>
+              <% if current_user.any_kind_of_delegate? || current_user.board_member? %>
```
Comment (jfly) [inline WcaOnRails/app/views/layouts/_navigation.html.erb]: I think we should use `current_user.can_view_delegate_matters?` here instead.

File: WcaOnRails/app/views/layouts/_navigation.html.erb [hunk 2]
```diff
@@ -136,13 +136,13 @@
...
                 <li class="divider"></li>
                 <li role="presentation" class="dropdown-header">
                   <%= t '.delegate' %>
                 </li>
                 <li><%= link_to t('.panel'), delegate_path %></li>
-                <li><%= link_to t('.new_competition'), new_competition_path %></li>
+                <li><%= link_to t('.new_competition'), new_competition_path if current_user.any_kind_of_delegate? %></li>
```
Comment (jfly) [inline WcaOnRails/app/views/layouts/_navigation.html.erb]: I think we should use `current_user.can_create_competitions?` here instead.

**Comment by AlbertoPdRF** [comment]
Thanks for the review @jfly, it should be good now.

**Comment by AlbertoPdRF** [comment]
Not sure why Travis failed, @viroulep will restart the job when he can.

**Comment by jfly** [comment]
Looking great!

**Pull request merged**